src/main/java/com/puppycrawl/tools/checkstyle/api/AuditEvent.java

```diff
@@ -37,11 +35,10 @@
  *
  * @see AuditListener
  */
-public final class AuditEvent
-    extends EventObject {
+public final class AuditEvent {
```

Comment:
referenced issue is not very clear on what actually is breaking change in this PR.

conceptually removal of extends in implements is breaking changes.
Doe sit make sense to make specific issue on this and merge in this release ? and final update in LocalizeMessage will be done in next release or when change is ready.

Reply:
Referenced issue was another way to resolve this issue, but I am recommending a different way. https://github.com/checkstyle/checkstyle/issues/7488#issuecomment-1402023707 says referenced issue is not good anyways.

> specific issue on this and merge in this release ? 

Yes, I recommend this.

> final update in LocalizeMessage will be done in next release or when change is ready

What final update are you referring to? Just adding documentation update referenced in https://github.com/checkstyle/checkstyle/issues/7488#issuecomment-1402023707 ?

Reply:
Yes, there will be no final update as it too big to break.

Reply:
Better to close #7488
And quickly create issue to explain what we breaking in this PR.

Reply:
> quickly create issue to explain what we breaking in this PR.

Done.

> Better to close https://github.com/checkstyle/checkstyle/issues/7488

Can't, it requires documentation update discussed at https://github.com/checkstyle/checkstyle/issues/7488#issuecomment-1402130818 .

Reply:
We should describe reason of removal `extends EventObject` in that issue. It might affect some plugins, developer should get details why we removed such extends/implements.

Reply:
I updated https://github.com/checkstyle/checkstyle/issues/12681 to contain all changes in API at one issue.

@rnveach , please squash both commits in one, as we try our best to be compile-able on each commit.

Reply:
> we try our best to be compile-able on each commit

Commits are compilable separate. It was just another CI failure that a serializable class had non-serializable fields.

> please squash both commits in one

Done.